### ISSUE TITLE
Wrong grad obtained from mlx.core.vjp

### ISSUE THREAD

**Issue by abdussamettrkr** [ml-explore/mlx#610]
**Describe the bug**
The wrong gradient is obtained from mlx.core.vjp. Removing the .tolist() method resolves the issue, and Jax behaves as expected. This bug does not occur in version 0.0.11. I think something wrong in broadcast.
 
**To Reproduce**
In version 0.1.0
```python
import mlx.core as mx
arr = mx.ones((3,3))
out_grad = mx.ones((3,3,3))
def func(arr):
    arr += mx.ones((3,3,3))
    cond = arr < 1
    cond.tolist()
    return arr**2

_, grad = mx.vjp(func,[arr], [out_grad])
print("Arr grad:", grad)
```
Prints:
Arr grad: [array([[0, 0, 0],
       [0, 0, 0],
       [0, 0, 0]], dtype=float32)]

In version 0.0.11 same code prints:
Arr grad: [array([[12, 12, 12],
       [12, 12, 12],
       [12, 12, 12]], dtype=float32)]

Jax implementation
```python
import jax
arr = jax.numpy.ones((3,3))
out_grad = jax.numpy.ones((3,3,3))
def func(arr):
    arr += jax.numpy.ones((3,3,3))
    cond = arr < 1
    cond.tolist()
    return arr**2

_, vjp_fn = jax.vjp(func, arr)
grad =vjp_fn(out_grad)
print("Arr grad:", grad)
```
Prints:
Arr grad: (Array([[12., 12., 12.],
       [12., 12., 12.],
       [12., 12., 12.]], dtype=float32),)

**Expected behavior**
Same result with Jax

 - OS Version:  MacOS 14.2.1
 - Version: 0.1.0

**Comment by awni** [comment]
Fixes are in #612
